### PR TITLE
Student service and student service tests

### PR DESCRIPTION
**Pull request by amankagarwal**

### CODE REVIEW

File: src/main/java/com/careerfair/q/model/db/Student.java [hunk 1]
```diff
@@ -6,9 +6,13 @@
...
 import com.google.cloud.firestore.annotation.Exclude;
 import com.google.cloud.firestore.annotation.PropertyName;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
 
 import java.util.List;
 
+@AllArgsConstructor
+@NoArgsConstructor
 public class Student {
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/model/db/Student.java:16]: Remove these if only for testing

File: src/main/java/com/careerfair/q/service/student/response/AddStudentResponse.java [hunk 2]
```diff
@@ -1,9 +1,7 @@
 package com.careerfair.q.service.student.response;
 
 import com.careerfair.q.model.db.Student;
-import lombok.EqualsAndHashCode;
 
-@EqualsAndHashCode(callSuper = true)
 public class AddStudentResponse extends StudentResponse {
     public AddStudentResponse(Student student) {
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/service/student/response/AddStudentResponse.java:6]: Add a blank line before constructor

File: src/main/java/com/careerfair/q/service/student/response/GetStudentResponse.java [hunk 3]
```diff
@@ -1,22 +1,9 @@
...
-import java.util.List;
-
-@Setter
-@Getter
-@EqualsAndHashCode(callSuper = true)
 public class GetStudentResponse extends StudentResponse {
-    @JsonProperty("employees")
-    private List<String> employees;
-
     public GetStudentResponse(Student student) {
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/service/student/response/GetStudentResponse.java:6]: Same here

File: src/main/java/com/careerfair/q/service/student/response/DeleteStudentResponse.java [hunk 4]
```diff
@@ -2,15 +2,11 @@
...
+import lombok.Data;
 
 import java.util.List;
 
-@Setter
-@Getter
-@EqualsAndHashCode(callSuper = true)
+@Data
 public class DeleteStudentResponse extends StudentResponse {
     @JsonProperty("student_id")
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/service/student/response/DeleteStudentResponse.java:11]: Same here

File: src/main/java/com/careerfair/q/service/student/response/StudentResponse.java [hunk 5]
```diff
@@ -4,11 +4,9 @@
...
 import com.google.cloud.Timestamp;
-import lombok.Getter;
-import lombok.Setter;
+import lombok.Data;
 
-@Getter
-@Setter
+@Data
 public abstract class StudentResponse {
     @JsonProperty("first_name")
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/service/student/response/StudentResponse.java:11]: Same here

File: src/main/java/com/careerfair/q/service/student/response/UpdateStudentResponse.java [hunk 6]
```diff
@@ -1,9 +1,7 @@
 package com.careerfair.q.service.student.response;
 
 import com.careerfair.q.model.db.Student;
-import lombok.EqualsAndHashCode;
 
-@EqualsAndHashCode(callSuper = true)
 public class UpdateStudentResponse extends StudentResponse {
     public UpdateStudentResponse(Student student) {
```
Comment (SarvagyaGupta) [inline src/main/java/com/careerfair/q/service/student/response/UpdateStudentResponse.java:6]: Same here

File: src/test/java/com/careerfair/q/service/student/implementation/StudentServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+        addStudentRequest.setLastName("l2");
+        addStudentRequest.setUniversityId("u2");
+        addStudentRequest.setMajor("m2");
+        addStudentRequest.setRole(Role.DS);
+        addStudentRequest.setBio("b2");
+        addStudentRequest.setEmail("s2@u2.edu");
+        addStudentRequest.setGpa(3.0);
+        addStudentRequest.setGraduationDate(Timestamp.ofTimeSecondsAndNanos(1592506825,
+                0));
+        addStudentRequest.setInternational(false);
```
Comment (SarvagyaGupta) [inline src/test/java/com/careerfair/q/service/student/implementation/StudentServiceTest.java:147]: Put into private method

File: src/test/java/com/careerfair/q/service/employee/implementation/EmployeeServiceTest.java [hunk 8]
```diff
@@ -125,19 +125,19 @@ public void testDeleteEmployee() {
     @Test
     public void testAddEmployee() {
         AddEmployeeRequest addEmployeeRequest = new AddEmployeeRequest();
-        addEmployeeRequest.setName("n1");
+        addEmployeeRequest.setName("n2");
         addEmployeeRequest.setCompanyId("c1");
         addEmployeeRequest.setBio("b1");
         addEmployeeRequest.setRole(Role.SWE);
         addEmployeeRequest.setEmail("e1@c1.com");
```
Comment (SarvagyaGupta) [inline src/test/java/com/careerfair/q/service/employee/implementation/EmployeeServiceTest.java:132]: Move to private method

File: src/test/java/com/careerfair/q/service/employee/implementation/EmployeeServiceTest.java [hunk 9]
```diff
@@ -65,24 +65,24 @@ public void testGetEmployeeWithEmail() {
...
-        updatedNameRequest.setCompanyId("c1");
-        updatedNameRequest.setRole(Role.SWE);
-        updatedNameRequest.setBio("b1");
-        updatedNameRequest.setEmail("e1@c1.com");
+        UpdateEmployeeRequest updateNameRequest = new UpdateEmployeeRequest();
+        updateNameRequest.setName("n2");
+        updateNameRequest.setCompanyId("c1");
+        updateNameRequest.setRole(Role.SWE);
+        updateNameRequest.setBio("b1");
+        updateNameRequest.setEmail("e1@c1.com");
```
Comment (SarvagyaGupta) [inline src/test/java/com/careerfair/q/service/employee/implementation/EmployeeServiceTest.java:73]: filter into private method

Comment (amankagarwal) [reply src/test/java/com/careerfair/q/service/employee/implementation/EmployeeServiceTest.java:73]: employee service, will make new branch and do it

**Pull request merged**